components/identity/org.wso2.carbon.identity.totp.endpoint/src/main/java/org/wso2/carbon/identity/totp/endpoint/TOTPTokenGeneratorEndpoint.java

```diff
@@ -0,0 +1,65 @@
+package org.wso2.carbon.identity.totp.endpoint;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.totp.TOTPManager;
+import org.wso2.carbon.identity.totp.exception.TOTPException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+
+@Path("/tokengen")
+public class TOTPTokenGeneratorEndpoint {
+
+	private static Log log = LogFactory.getLog(TOTPTokenGeneratorEndpoint.class);
+
+	@POST
+	@Path("/local")
+	@Consumes("application/x-www-form-urlencoded")
+	@Produces("application/json")
+	public Response generateBuyUsername(@Context HttpServletRequest request) {
+
+		TOTPManager totpManager = (TOTPManager) PrivilegedCarbonContext.getThreadLocalCarbonContext().getOSGiService
```

Comment:
Use the CarbonContext instead of PrivilegedCarbonContext here. Also the getOSGiService(Class) is deprecated now, use getOSGiService(Class,Hashtable)